### ISSUE TITLE
Delegating from form object causes `ArgumentError` with 0.2.4

### ISSUE THREAD

**Issue by mjgiarlo** [trailblazer/reform-rails#99]
I upgraded a Rails app that uses reform-rails from 0.2.3 to 0.2.4 today, and RSpec builds will no longer run due to an apparent delegation-related bug. The app in particular uses Ruby 3.1.2 and Rails 7.0.4.3. Pinning back to 0.2.3, these errors go away.

Example error:

```
An error occurred while loading ./spec/forms/create_collection_form_spec.rb.
Failure/Error: delegate :persisted?, :to_param, to: :collection

ArgumentError:
  wrong number of arguments (given 3, expected 1)
# ./app/forms/draft_collection_form.rb:97:in `<class:DraftCollectionForm>'
# ./app/forms/draft_collection_form.rb:4:in `<top (required)>'
# /media/mjg/44aee93d-a48a-4ad3-ae90-4b084bb04b69/rvm/gems/ruby-3.1.2@h2/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/k
ernel_require.rb:32:in `require'
# /media/mjg/44aee93d-a48a-4ad3-ae90-4b084bb04b69/rvm/gems/ruby-3.1.2@h2/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/k
ernel_require.rb:32:in `require'
# /media/mjg/44aee93d-a48a-4ad3-ae90-4b084bb04b69/rvm/gems/ruby-3.1.2@h2/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:30:in `require'
# ./app/forms/create_collection_form.rb:4:in `<top (required)>'
# /media/mjg/44aee93d-a48a-4ad3-ae90-4b084bb04b69/rvm/gems/ruby-3.1.2@h2/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/k
ernel_require.rb:32:in `require'
# /media/mjg/44aee93d-a48a-4ad3-ae90-4b084bb04b69/rvm/gems/ruby-3.1.2@h2/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/k
ernel_require.rb:32:in `require'
# /media/mjg/44aee93d-a48a-4ad3-ae90-4b084bb04b69/rvm/gems/ruby-3.1.2@h2/gems/zeitwerk-2.6.8/lib/zeitwerk/kernel.rb:30:in `require'
# ./spec/forms/create_collection_form_spec.rb:5:in `<top (required)>'
```

Here's the form object in question: https://github.com/sul-dlss/happy-heron/blob/main/app/forms/draft_collection_form.rb#L97

**Comment by apotonick** [comment]
Thanks Michael for this detailed bug report! :beers: 
I think this is related to https://dev.to/trailblazer/dont-mix-forwardable-and-activesupportdelegate-5hmm  @richardboehme !

**Comment by richardboehme** [comment]
@mjgiarlo I fixed this and released 0.2.5! Hopefully everything works as expected now :sweat_smile:

**Comment by mjgiarlo** [comment]
@richardboehme Upgraded to 0.2.5 and it works like a charm. Thank you!
